link/src/main/java/com/stripe/android/link/account/LinkAccountManager.kt

```
@@ -182,83 +151,23 @@ internal class LinkAccountManager @Inject constructor(
-                        // Try fetching the user account with the stored cookie, then retry API call
-                        lookupConsumer(null).fold(
-                            onSuccess = {
-                                it?.let { updatedAccount ->
-                                    apiCall(updatedAccount.clientSecret)
-                                }
-                            },
-                            onFailure = {
-                                Result.failure(it)
-                            }
-                        )
-                    } else {
-                        Result.failure(it)
-                    }
-                }
-            )
-        } ?: Result.failure(
-            IllegalStateException("User not signed in")
-        )
-
-    /**
-     * Logs the current consumer out.
-     *
-     * Regardless of the result of the API call, the local cookie is deleted and the current account
-     * is cleared. This will effectively log the user out, so there's no need to wait for the result
-     * of this call to consider it done.
-     */
-    fun logout() =
-        linkAccount.value?.let { account ->
-            val cookie = cookie()
-            cookieStore.logout(account.email)
-            userHasLoggedOut = true
-            _linkAccount.value = null
-            val publishableKey = consumerPublishableKey
-            consumerPublishableKey = null
-            GlobalScope.launch {
-                linkRepository.logout(account.clientSecret, publishableKey, cookie)
-            }
-        }
-
-    /**
-     * Whether the user has logged out from any account during this session, or the last logout was
-     * from the [email] passed as parameter, even if in a previous session.
-     */
-    fun hasUserLoggedOut(email: String?) = userHasLoggedOut ||
-        (email?.let { cookieStore.isEmailLoggedOut(it) } ?: false)
+    } ?: Result.failure(
+        IllegalStateException("User not signed in")
+    )
```

Not to be resolved here, but it’d be valuable to learn if this (and other situations like it with silent failures) occur frequently. Basically, tracking expectations that weren’t fulfilled.

---

Yeah, agreed. We should never see this in practice. We do have some analytics events we can attach this sort of thing to. Might be worth doing.